src/LogExpert.Core/Classes/Filter/FilterPipe.cs

```diff
@@ -191,5 +191,29 @@ private void OnClosed ()
         Closed?.Invoke(this, EventArgs.Empty);
     }
 
+    public void Dispose ()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+
+    protected virtual void Dispose (bool disposing)
+    {
+        if (!_disposed)
+        {
+            if (disposing)
+            {
+                _writer?.Dispose();
+            }
+            _disposed = true;
+        }
+    }
+
+    ~FilterPipe ()
```

Comment:
Destructors (finalizers) do not need to be implemented it in 99.99% of the cases, https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/finalizers.

Reply:
Done